RecoTracker/LSTCore compilation warning in UBSAN and DBG IBs

loop not unrolled compilation warnings in UBSAN and DBG IBs 
Near
[src/RecoTracker/LSTCore/src/alpaka/TrackCandidate.h](https://github.com/cms-sw/cmssw/blob/CMSSW_16_0_UBSAN_X_2025-12-17-2300/RecoTracker/LSTCore/src/alpaka/TrackCandidate.h#L668)
 ```BASH
>> Compiling alpaka/rocm src/RecoTracker/LSTCore/src/alpaka/LSTEvent.dev.cc
/data/cmsbld/jenkins/workspace/build-any-ib/w/el8_amd64_gcc13/external/rocm/7.1.0-3368aff4c211469d06c08ddf03fbf8f5/bin/hipcc -c -DCMS_MICRO_ARCH='x86-64-v3' -DGNU_GCC -D_GNU_SOURCE -DTBB_USE_GLIBCXX_VERSION=130400 -DTBB_SUPPRESS_DEPRECATED_MESSAGES -DTBB_PREVIEW_RESUMABLE_TASKS=1 -DTBB_PREVIEW_TASK_GROUP_EXTENSIONS=1 -D__HIP_PLATFORM_HCC__ -D__HIP_PLATFORM_AMD__ -DBOOST_SPIRIT_THREADSAFE -DPHOENIX_THREADSAFE -DBOOST_MATH_DISABLE_STD_FPCLASSIFY -DBOOST_UUID_RANDOM_PROVIDER_FORCE_POSIX -DBOOST_MPL_IGNORE_PARENTHESES_WARNING -DCMSSW_GIT_HASH='CMSSW_16_0_UBSAN_X_2025-12-17-2300' -DPROJECT_NAME='CMSSW' -DPROJECT_VERSION='CMSSW_16_0_UBSAN_X_2025-12-17-2300' -Isrc -I/data/cmsbld/jenkins/workspace/build-any-ib/w/el8_amd64_gcc13/external/pcre/8.43-6d98fda3bfd074ebb583e2d6a2c75d25/include -isystem/data/cmsbld/jenkins/workspace/build-any-ib/w/el8_amd64_gcc13/external/boost/1.80.0-b819d3899535842b3b08dcd6a725af1a/include -I/data/cmsbld/jenkins/workspace/build-any-ib/w/el8_amd64_gcc13/external/bz2lib/1.0.6-d113e1c6278c07eeaff5f84db9548446/include -isystem/data/cmsbld/jenkins/workspace/build-any-ib/w/el8_amd64_gcc13/external/clhep/2.4.7.1-caf57578c24c3884afdedd94a88b58d7/include -I/data/cmsbld/jenkins/workspace/build-any-ib/w/el8_amd64_gcc13/external/gsl/2.6-9011a41928244b609ca4c22c439b3fef/include -I/data/cmsbld/jenkins/workspace/build-any-ib/w/el8_amd64_gcc13/external/libuuid/2.34-5ba7a8abfc0c5fecdc448cca360c25ff/include -isystem/data/cmsbld/jenkins/workspace/build-any-ib/w/el8_amd64_gcc13/external/rocm/7.1.0-3368aff4c211469d06c08ddf03fbf8f5/include -isystem/data/cmsbld/jenkins/workspace/build-any-ib/w/el8_amd64_gcc13/lcg/root/6.36.07-e0ce66fada32a38695cf14867f0ab7a8/include -isystem/data/cmsbld/jenkins/workspace/build-any-ib/w/el8_amd64_gcc13/external/tbb/v2022.3.0-88eb7be4ee320d604a798a914aea6359/include -I/data/cmsbld/jenkins/workspace/build-any-ib/w/el8_amd64_gcc13/external/xz/5.6.4-b9c4ffbc390ed320a5d57fd552e29a05/include -I/data/cmsbld/jenkins/workspace/build-any-ib/w/el8_amd64_gcc13/external/zlib/1.2.13-589f6bb51bbeba38a7adf5a10ea8a093/include -I/data/cmsbld/jenkins/workspace/build-any-ib/w/el8_amd64_gcc13/external/alpaka/2.0.0-8493f1d11d0378dc14d6ea6ecfc69ac5/include -I/data/cmsbld/jenkins/workspace/build-any-ib/w/el8_amd64_gcc13/external/eigen/3bb6a48d8c171cf20b5f8e48bfb4e424fbd4f79e-95c02b8a883b2934decb8bb53ff9b486/include -I/data/cmsbld/jenkins/workspace/build-any-ib/w/el8_amd64_gcc13/external/eigen/3bb6a48d8c171cf20b5f8e48bfb4e424fbd4f79e-95c02b8a883b2934decb8bb53ff9b486/include/eigen3 -I/data/cmsbld/jenkins/workspace/build-any-ib/w/el8_amd64_gcc13/external/fmt/10.2.1-31d67b0504b4ba2262f03d3c5cad83c1/include -I/data/cmsbld/jenkins/workspace/build-any-ib/w/el8_amd64_gcc13/external/md5/1.0.0-26057075013e190e56dad37d35219376/include -I/data/cmsbld/jenkins/workspace/build-any-ib/w/el8_amd64_gcc13/external/OpenBLAS/0.3.27-da4a3c2bb8ae43f3913a4a44acdb1b50/include -I/data/cmsbld/jenkins/workspace/build-any-ib/w/el8_amd64_gcc13/external/tinyxml2/6.2.0-67924ead96ecb4e69aad321b767979a5/include -O3 -pthread -pipe -Werror=main -Werror=pointer-arith -Werror=overlength-strings -Wno-vla -Werror=overflow -std=c++20 -ftree-vectorize -Werror=array-bounds -Werror=type-limits -fvisibility-inlines-hidden -fno-math-errno --param vect-max-version-for-alias-checks=50 -Xassembler --compress-debug-sections -Wno-error=array-bounds -Warray-bounds -fuse-ld=bfd -march=x86-64-v3 -felide-constructors -fmessage-length=0 -Wall -Wno-long-long -Wreturn-type -Wextra -Wpessimizing-move -Wclass-memaccess -Wno-cast-function-type -Wno-unused-but-set-parameter -Wno-ignored-qualifiers -Wno-unused-parameter -Wunused -Wparentheses -Werror=return-type -Werror=unused-value -Werror=unused-label -Werror=address -Werror=format -Werror=sign-compare -Werror=write-strings -Werror=delete-non-virtual-dtor -Werror=strict-aliasing -Werror=narrowing -Werror=reorder -Werror=unused-variable -Werror=conversion-null -Wnon-virtual-dtor -Werror=switch -fdiagnostics-show-option -Wno-unused-local-typedefs -Wno-attributes -Wno-psabi -Wno-c99-extensions -Wno-c++11-narrowing -D__STRICT_ANSI__ -Wno-unused-private-field -Wno-unknown-pragmas -Wno-unused-command-line-argument -Wno-unknown-warning-option -ftemplate-depth=512 -Wno-error=potentially-evaluated-expression -Wno-tautological-type-limit-compare -Wno-vla-cxx-extension -fsized-deallocation -DEIGEN_DONT_PARALLELIZE -DEIGEN_MAX_ALIGN_BYTES=64 -DALPAKA_DEFAULT_HOST_MEMORY_ALIGNMENT=128 -DALPAKA_DISABLE_VENDOR_RNG -DALPAKA_HAS_STD_ATOMIC_REF -Wno-error=unused-variable -DALPAKA_ACC_GPU_HIP_ENABLED -DALPAKA_ACC_GPU_HIP_ONLY_MODE -DALPAKA_HOST_ONLY -DBOOST_DISABLE_ASSERTS -g --rocm-path=/data/cmsbld/jenkins/workspace/build-any-ib/w/el8_amd64_gcc13/external/rocm/7.1.0-3368aff4c211469d06c08ddf03fbf8f5 --offload-arch=gfx908:sramecc+ --offload-arch=gfx90a:sramecc+ --offload-arch=gfx942:sramecc+ --offload-arch=gfx1030 --offload-arch=gfx1100 --offload-arch=gfx1102 -fgpu-rdc -DALPAKA_ACC_GPU_HIP_ENABLED -DALPAKA_ACC_GPU_HIP_ONLY_MODE -UALPAKA_HOST_ONLY -fPIC -MMD -MF tmp/el8_amd64_gcc13/src/RecoTracker/LSTCore/src/alpaka/RecoTrackerLSTCoreROCmAsync/LSTEvent.dev.cc.d src/RecoTracker/LSTCore/src/alpaka/LSTEvent.dev.cc -o tmp/el8_amd64_gcc13/src/RecoTracker/LSTCore/src/alpaka/RecoTrackerLSTCoreROCmAsync/LSTEvent.dev.cc.o
In file included from src/RecoTracker/LSTCore/src/alpaka/LSTEvent.dev.cc:14:
  src/RecoTracker/LSTCore/src/alpaka/TrackCandidate.h:668:7: warning: loop not unrolled: the optimizer was unable to perform the requested transformation; the transformation might be disabled or specified as part of an unsupported transformation ordering [-Wpass-failed=transform-warning]
   668 |       for (int baseHitIndex = 0; baseHitIndex < Params_T5::kHits; ++baseHitIndex) {
      |       ^
  src/RecoTracker/LSTCore/src/alpaka/TrackCandidate.h:668:7: warning: loop not unrolled: the optimizer was unable to perform the requested transformation; the transformation might be disabled or specified as part of an unsupported transformation ordering [-Wpass-failed=transform-warning]
   src/RecoTracker/LSTCore/src/alpaka/TrackCandidate.h:668:7: warning: loop not unrolled: the optimizer was unable to perform the requested transformation; the transformation might be disabled or specified as part of an unsupported transformation ordering [-Wpass-failed=transform-warning]
   src/RecoTracker/LSTCore/src/alpaka/TrackCandidate.h:668:7: warning: loop not unrolled: the optimizer was unable to perform the requested transformation; the transformation might be disabled or specified as part of an unsupported transformation ordering [-Wpass-failed=transform-warning]
   src/RecoTracker/LSTCore/src/alpaka/TrackCandidate.h:668:7: warning: loop not unrolled: the optimizer was unable to perform the requested transformation; the transformation might be disabled or specified as part of an unsupported transformation ordering [-Wpass-failed=transform-warning]
 5 warnings generated when compiling for gfx1030.
In file included from src/RecoTracker/LSTCore/src/alpaka/LSTEvent.dev.cc:14:
  src/RecoTracker/LSTCore/src/alpaka/TrackCandidate.h:668:7: warning: loop not unrolled: the optimizer was unable to perform the requested transformation; the transformation might be disabled or specified as part of an unsupported transformation ordering [-Wpass-failed=transform-warning]
   668 |       for (int baseHitIndex = 0; baseHitIndex < Params_T5::kHits; ++baseHitIndex) {
      |       ^
  src/RecoTracker/LSTCore/src/alpaka/TrackCandidate.h:668:7: warning: loop not unrolled: the optimizer was unable to perform the requested transformation; the transformation might be disabled or specified as part of an unsupported transformation ordering [-Wpass-failed=transform-warning]
   src/RecoTracker/LSTCore/src/alpaka/TrackCandidate.h:668:7: warning: loop not unrolled: the optimizer was unable to perform the requested transformation; the transformation might be disabled or specified as part of an unsupported transformation ordering [-Wpass-failed=transform-warning]
   src/RecoTracker/LSTCore/src/alpaka/TrackCandidate.h:668:7: warning: loop not unrolled: the optimizer was unable to perform the requested transformation; the transformation might be disabled or specified as part of an unsupported transformation ordering [-Wpass-failed=transform-warning]
   src/RecoTracker/LSTCore/src/alpaka/TrackCandidate.h:668:7: warning: loop not unrolled: the optimizer was unable to perform the requested transformation; the transformation might be disabled or specified as part of an unsupported transformation ordering [-Wpass-failed=transform-warning]
 5 warnings generated when compiling for gfx1100.
In file included from src/RecoTracker/LSTCore/src/alpaka/LSTEvent.dev.cc:14:
  src/RecoTracker/LSTCore/src/alpaka/TrackCandidate.h:668:7: warning: loop not unrolled: the optimizer was unable to perform the requested transformation; the transformation might be disabled or specified as part of an unsupported transformation ordering [-Wpass-failed=transform-warning]
   668 |       for (int baseHitIndex = 0; baseHitIndex < Params_T5::kHits; ++baseHitIndex) {
      |       ^
  src/RecoTracker/LSTCore/src/alpaka/TrackCandidate.h:668:7: warning: loop not unrolled: the optimizer was unable to perform the requested transformation; the transformation might be disabled or specified as part of an unsupported transformation ordering [-Wpass-failed=transform-warning]
   src/RecoTracker/LSTCore/src/alpaka/TrackCandidate.h:668:7: warning: loop not unrolled: the optimizer was unable to perform the requested transformation; the transformation might be disabled or specified as part of an unsupported transformation ordering [-Wpass-failed=transform-warning]
   src/RecoTracker/LSTCore/src/alpaka/TrackCandidate.h:668:7: warning: loop not unrolled: the optimizer was unable to perform the requested transformation; the transformation might be disabled or specified as part of an unsupported transformation ordering [-Wpass-failed=transform-warning]
   src/RecoTracker/LSTCore/src/alpaka/TrackCandidate.h:668:7: warning: loop not unrolled: the optimizer was unable to perform the requested transformation; the transformation might be disabled or specified as part of an unsupported transformation ordering [-Wpass-failed=transform-warning]
 5 warnings generated when compiling for gfx1102.
In file included from src/RecoTracker/LSTCore/src/alpaka/LSTEvent.dev.cc:14:
  src/RecoTracker/LSTCore/src/alpaka/TrackCandidate.h:668:7: warning: loop not unrolled: the optimizer was unable to perform the requested transformation; the transformation might be disabled or specified as part of an unsupported transformation ordering [-Wpass-failed=transform-warning]
   668 |       for (int baseHitIndex = 0; baseHitIndex < Params_T5::kHits; ++baseHitIndex) {
      |       ^
  src/RecoTracker/LSTCore/src/alpaka/TrackCandidate.h:668:7: warning: loop not unrolled: the optimizer was unable to perform the requested transformation; the transformation might be disabled or specified as part of an unsupported transformation ordering [-Wpass-failed=transform-warning]
   src/RecoTracker/LSTCore/src/alpaka/TrackCandidate.h:668:7: warning: loop not unrolled: the optimizer was unable to perform the requested transformation; the transformation might be disabled or specified as part of an unsupported transformation ordering [-Wpass-failed=transform-warning]
   src/RecoTracker/LSTCore/src/alpaka/TrackCandidate.h:668:7: warning: loop not unrolled: the optimizer was unable to perform the requested transformation; the transformation might be disabled or specified as part of an unsupported transformation ordering [-Wpass-failed=transform-warning]
   src/RecoTracker/LSTCore/src/alpaka/TrackCandidate.h:668:7: warning: loop not unrolled: the optimizer was unable to perform the requested transformation; the transformation might be disabled or specified as part of an unsupported transformation ordering [-Wpass-failed=transform-warning]
 5 warnings generated when compiling for gfx908.
In file included from src/RecoTracker/LSTCore/src/alpaka/LSTEvent.dev.cc:14:
  src/RecoTracker/LSTCore/src/alpaka/TrackCandidate.h:668:7: warning: loop not unrolled: the optimizer was unable to perform the requested transformation; the transformation might be disabled or specified as part of an unsupported transformation ordering [-Wpass-failed=transform-warning]
   668 |       for (int baseHitIndex = 0; baseHitIndex < Params_T5::kHits; ++baseHitIndex) {
      |       ^
  src/RecoTracker/LSTCore/src/alpaka/TrackCandidate.h:668:7: warning: loop not unrolled: the optimizer was unable to perform the requested transformation; the transformation might be disabled or specified as part of an unsupported transformation ordering [-Wpass-failed=transform-warning]
   src/RecoTracker/LSTCore/src/alpaka/TrackCandidate.h:668:7: warning: loop not unrolled: the optimizer was unable to perform the requested transformation; the transformation might be disabled or specified as part of an unsupported transformation ordering [-Wpass-failed=transform-warning]
   src/RecoTracker/LSTCore/src/alpaka/TrackCandidate.h:668:7: warning: loop not unrolled: the optimizer was unable to perform the requested transformation; the transformation might be disabled or specified as part of an unsupported transformation ordering [-Wpass-failed=transform-warning]
   src/RecoTracker/LSTCore/src/alpaka/TrackCandidate.h:668:7: warning: loop not unrolled: the optimizer was unable to perform the requested transformation; the transformation might be disabled or specified as part of an unsupported transformation ordering [-Wpass-failed=transform-warning]
 5 warnings generated when compiling for gfx90a.
In file included from src/RecoTracker/LSTCore/src/alpaka/LSTEvent.dev.cc:14:
  src/RecoTracker/LSTCore/src/alpaka/TrackCandidate.h:668:7: warning: loop not unrolled: the optimizer was unable to perform the requested transformation; the transformation might be disabled or specified as part of an unsupported transformation ordering [-Wpass-failed=transform-warning]
   668 |       for (int baseHitIndex = 0; baseHitIndex < Params_T5::kHits; ++baseHitIndex) {
      |       ^
  src/RecoTracker/LSTCore/src/alpaka/TrackCandidate.h:668:7: warning: loop not unrolled: the optimizer was unable to perform the requested transformation; the transformation might be disabled or specified as part of an unsupported transformation ordering [-Wpass-failed=transform-warning]
   src/RecoTracker/LSTCore/src/alpaka/TrackCandidate.h:668:7: warning: loop not unrolled: the optimizer was unable to perform the requested transformation; the transformation might be disabled or specified as part of an unsupported transformation ordering [-Wpass-failed=transform-warning]
   src/RecoTracker/LSTCore/src/alpaka/TrackCandidate.h:668:7: warning: loop not unrolled: the optimizer was unable to perform the requested transformation; the transformation might be disabled or specified as part of an unsupported transformation ordering [-Wpass-failed=transform-warning]
   src/RecoTracker/LSTCore/src/alpaka/TrackCandidate.h:668:7: warning: loop not unrolled: the optimizer was unable to perform the requested transformation; the transformation might be disabled or specified as part of an unsupported transformation ordering [-Wpass-failed=transform-warning]
 5 warnings generated when compiling for gfx942.
```

Full Logs: [here](https://cmssdt.cern.ch/SDT/cgi-bin/buildlogs/el8_amd64_gcc13/CMSSW_16_0_UBSAN_X_2025-12-17-2300/RecoTracker/LSTCore)

--- a/RecoTracker/LSTCore/src/alpaka/TrackCandidate.h
+++ b/RecoTracker/LSTCore/src/alpaka/TrackCandidate.h
@@ -663,8 +663,10 @@ namespace ALPAKA_ACCELERATOR_NAMESPACE::lst {
 
       bool hit0InBase = false;
       bool hit1InBase = false;
-
+      // Suppress compiler warning with HIP backend
+#ifndef ALPAKA_ACC_GPU_HIP_ENABLED
       CMS_UNROLL_LOOP
+#endif
       for (int baseHitIndex = 0; baseHitIndex < Params_T5::kHits; ++baseHitIndex) {
         const unsigned int baseHit = quintuplets.hitIndices()[refT5Index][baseHitIndex];
         if (candidateHit0 == baseHit)

